src/foam/nanos/approval/RoleApprovalRequest.js

```diff
@@ -0,0 +1,307 @@
+        if ( status === foam.nanos.approval.ApprovalStatus.REQUESTED ) {
+          return foam.u2.DisplayMode.HIDDEN;
+        }
+
+        return foam.u2.DisplayMode.RO;
+      }
+    },
+    {
+      class: 'String',
+      name: 'pendingApproval',
+      label: 'Approver',
+      section: 'requestDetails',
+      transient: true,
+      value: 'Pending',
+      visibility: function(status) {
+        if ( status === foam.nanos.approval.ApprovalStatus.REQUESTED ) {
+          return foam.u2.DisplayMode.RO;
+        }
+        return foam.u2.DisplayMode.HIDDEN;
+      },
+      documentation: `
+        This string will be used to show that the approver is pending without
+        altering the value of Approver. It is also transient.
+      `
+    },
+    {
+      class: 'Enum',
+      of: 'foam.nanos.ruler.Operations',
+      name: 'operation',
+      label: 'Action',
+      section: 'requestDetails',
+      visibility: function(operation) {
+        return operation ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'initiatingUser',
+      label: 'Requestor',
+      tableCellFormatter: function(initiatingUser) {
+        let self = this;
+        this.__subSubContext__.userDAO.find(initiatingUser).then(user => {
+          self.add(user ? user.toSummary() : `User #${initiatingUser}`);
+        });
+      },
+      section: 'requestDetails',
+      visibility: function(initiatingUser) {
+        return initiatingUser ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Boolean',
+      name: 'isFulfilled',
+      visibility: 'HIDDEN'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'toSummary',
+      code: function() {
+        return `(${this.classification}) ${this.operation}`;
+      }
+    },
+    {
+      name: 'validate',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      javaCode: `
+      Logger logger = (Logger) x.get("logger");
+      DAO dao = (DAO) x.get(getDaoKey());
+      if ( dao == null ) {
+        logger.error(this.getClass().getSimpleName(), "DaoKey not found", getDaoKey());
+        throw new RuntimeException("Invalid dao key for the approval request object.");
+      }
+
+      if ( getOperation() != Operations.CREATE ){
+        FObject obj = dao.inX(x).find(getObjId());
+        if ( obj == null ) {
+          logger.error(this.getClass().getSimpleName(), "ObjId not found", getObjId());
+          throw new RuntimeException("Invalid object id.");
+        }
+      }
+      `
+    },
+  ],
+
+  messages: [
+    {
+      name: 'SUCCESS_APPROVED',
+      message: 'You have successfully approved this request.'
+    },
+    {
+      name: 'SUCCESS_REJECTED',
+      message: 'You have successfully rejected this request.'
+    },
+    {
+      name: 'APPROVER_PENDING',
```

Comment:
for the enum itself please use REQUESTED, you can make the label, message 'Pending'

Reply:
Done.